src/archive.cpp

```diff
@@ -261,6 +261,10 @@ namespace zim
     // If we search for prefix a, we must return 0/4
     // A findx with a will return 0
     // A find with b will return 4
+
+    // If path is a namespace url "A/" or "/A/" remove trailing slash
+    if(path.size() <= 3 && path.back() == '/') path.pop_back();
```

Comment:
This workaround should be employed only if the archive uses the old namespace scheme.

Besides, the case of the empty path will cause problems (the old code was not protected against that corner case too). I think we can add an `ASSERT` that `path` is not empty.

Reply:
@veloman-yunkan so I should modify this to:
```c++
Archive::EntryRange<EntryOrder::pathOrder> Archive::findByPath(std::string path) const
{
    entry_index_t begin_idx, end_idx;
    if (m_impl->hasNewNamespaceScheme()) {
      ...
    } else {
      ASSERT(path.empty(), !=, true)
      if(path.size() <= 3 && path.back() == '/') path.pop_back();
      ...
    }
    return Archive::EntryRange<EntryOrder::pathOrder>(m_impl, begin_idx.v, end_idx.v);
}
```
Is this the correct approach?

Reply:
Let the `ASSERT` be outside/above the `if` statement

Reply:
It would be better to handle a empty `path` as a special case and returning the same Range than `iterByPath`.

I've used a lot of ASSERT in the past but we should not.
The ASSERT should be use for assertion (things that must be always true. If not, this is a bug) ASSERT may be removed in the future in release mode.

Here we are checking a value coming from the user. We must check it "normally" and throw a correct exception.

Reply:
@mgautierfr That makes sense. For this particular issue, I guess having a separate `if(path.empty())` will be fine.
```c++
Archive::EntryRange<EntryOrder::pathOrder> Archive::findByPath(std::string path) const
{
    entry_index_t begin_idx, end_idx;
    if(path.empty()){
       begin_idx = m_impl->getStartUserEntry();
       end_idx = m_impl->getEndUserEntry()
    } else if (m_impl->hasNewNamespaceScheme()) {
      ...
    } else {
      if(path.size() <= 3 && path.back() == '/') path.pop_back();
      ...
    }
    return Archive::EntryRange<EntryOrder::pathOrder>(m_impl, begin_idx.v, end_idx.v);
}
```
Is this the change you are suggesting?

Reply:
Yes.

Can you add test for `/` path also please ? (as if the path is empty)